### PR TITLE
Add comment for why use the user provided ctx in client side stream

### PR DESCRIPTION
**Pull request by menghanl**
@tschottdorf

### CODE REVIEW

**Comment by tschottdorf** [comment]
LGTM, thanks.

File: transport/http2_client.go [hunk 1]
```diff
@@ -252,6 +252,9 @@ func (t *http2Client) newStream(ctx context.Context, callHdr *CallHdr) *Stream {
 	s.windowHandler = func(n int) {
 		t.updateWindow(s, uint32(n))
 	}
+	// The client side stream context should have exactly the same life cycle with the user provided context.
+	// That means, s.ctx should be read-only. And s.ctx is done iff ctx is done.
```
Comment (Thomasdezeeuw) [inline transport/http2_client.go]: Type-o: iff -> if.

Comment (menghanl) [inline transport/http2_client.go]: By iff I mean "if and only if".

Comment (Thomasdezeeuw) [inline transport/http2_client.go]: Ah never seen that before, good to know, thanks.

**Pull request merged**